### PR TITLE
Function for creating inso app registration

### PR DESCRIPTION
**Pull request by scottmelhop**

### CODE REVIEW

File: AddCogniteInsoAppRegistration.ps1 [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                type = "Scope"
+            }
+        )
+    }
+
+    # Combine resource access objects and output as Json to manifest.json
+    @($GRAPH_RESOURCE_ACCESS, $CDF_RESOURCE_ACCESS) | ConvertTo-Json -Depth 5 | Out-File $MANIFEST_FILE
+
+    # Set callback path used by oauth2 proxy
+    $CALLBACK_PATH = "/oauth2/callback"
```
Comment (sdorheim) [inline AddCogniteInsoAppRegistration.ps1]: wouldn't it look better if you removed the first / and put it in the create-command?

File: AddCogniteInsoAppRegistration.ps1 [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            @{
+                id = "37f7f235-527c-4136-accd-4a02d197296e"
+                type = "Scope"
+            },
+            @{
+                id = "64a6cdd6-aab1-4aaf-94b8-3cc8405e90d0"
+                type = "Scope"
+            },
+            @{
+                id = "14dad69e-099b-42c9-810b-d002981feec1"
```
Comment (sdorheim) [inline AddCogniteInsoAppRegistration.ps1]: I asume these to be correct

File: AddCogniteInsoAppRegistration.ps1 [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            --display-name $AppName `
+            --available-to-other-tenants false `
+            --oauth2-allow-implicit-flow false `
+            --reply-urls "http://localhost:4180$CALLBACK_PATH" `
+            "https://$Customer-test.cogniteapp.com$CALLBACK_PATH" `
+            "https://$Customer.cogniteapp.com$CALLBACK_PATH" `
+            --required-resource-accesses `@manifest.json | ConvertFrom-Json
+
+    # Create or update Client Secret
+    $CLIENT_SECRET = az ad app credential reset --id $NEW_APP_REGISTRATION.appId --credential-description cogniteapps --end-date "2099-01-01" | ConvertFrom-Json
```
Comment (sdorheim) [inline AddCogniteInsoAppRegistration.ps1]: do we want to remove the old credentials in all cases?

File: AddCogniteInsoAppRegistration.ps1 [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    # Create or update App Registration
+    $NEW_APP_REGISTRATION = az ad app create `
+            --display-name $AppName `
+            --available-to-other-tenants false `
+            --oauth2-allow-implicit-flow false `
+            --reply-urls "http://localhost:4180$CALLBACK_PATH" `
+            "https://$Customer-test.cogniteapp.com$CALLBACK_PATH" `
+            "https://$Customer.cogniteapp.com$CALLBACK_PATH" `
+            --required-resource-accesses `@manifest.json | ConvertFrom-Json
```
Comment (sdorheim) [inline AddCogniteInsoAppRegistration.ps1]: use the $MANIFEST_FILE variable?

File: AddCogniteInsoAppRegistration.ps1 (hunk 3 above)
Comment (scottmelhop) [reply AddCogniteInsoAppRegistration.ps1]: no, i was going to look into that later.

File: AddCogniteInsoAppRegistration.ps1 (hunk 2 above)
Comment (scottmelhop) [reply AddCogniteInsoAppRegistration.ps1]: Yeap, but I could comment them to make them more explicit in the script incase someone reads

File: AddCogniteInsoAppRegistration.ps1 (hunk 1 above)
Comment (scottmelhop) [reply AddCogniteInsoAppRegistration.ps1]: You say potet, I say potato :)

File: AddCogniteInsoAppRegistration.ps1 (hunk 4 above)
Comment (scottmelhop) [reply AddCogniteInsoAppRegistration.ps1]: 👍🏿

**Review by sdorheim — APPROVED** [review]
looks good as a start :)

**Pull request merged**